apps/interop-rxjs-signal/src/app/list/photos.store.ts

```diff
@@ -30,106 +36,90 @@ const initialState: PhotoState = {
   error: '',
 };
 
-@Injectable()
-export class PhotoStore
-  extends ComponentStore<PhotoState>
-  implements OnStoreInit, OnStateInit
-{
-  private photoService = inject(PhotoService);
+const getSavedState = (): PhotoState => {
+  const savedJSONState = localStorage.getItem(PHOTO_STATE_KEY);
+  const parsedState = savedJSONState && JSON.parse(savedJSONState);
+  return parsedState
+    ? { ...defaultState, search: parsedState.search, page: parsedState.page }
+    : defaultState;
+};
 
-  private readonly photos$ = this.select((s) => s.photos);
-  private readonly search$ = this.select((s) => s.search);
-  private readonly page$ = this.select((s) => s.page);
-  private readonly pages$ = this.select((s) => s.pages);
-  private readonly error$ = this.select((s) => s.error);
-  private readonly loading$ = this.select((s) => s.loading);
+@Injectable({ providedIn: 'root' })
+export class PhotoStore {
+  private readonly photoService = inject(PhotoService);
 
-  private readonly endOfPage$ = this.select(
-    this.page$,
-    this.pages$,
-    (page, pages) => page === pages
-  );
+  // Signals
+  readonly state = signal(getSavedState());
+  readonly photos = computed(() => this.state().photos);
+  readonly page = computed(() => this.state().page);
+  readonly pages = computed(() => this.state().pages);
+  readonly error = computed(() => this.state().error);
+  readonly loading = computed(() => this.state().loading);
+  readonly endOfPage = computed(() => this.page() === this.pages());
+
+  // Sources
+  readonly searchControl = new FormControl({
+    value: getSavedState().search,
+    disabled: false,
+  });
 
-  readonly vm$ = this.select(
-    {
-      photos: this.photos$,
-      search: this.search$,
-      page: this.page$,
-      pages: this.pages$,
-      endOfPage: this.endOfPage$,
-      loading: this.loading$,
-      error: this.error$,
-    },
-    { debounce: true }
+  readonly search$ = this.searchControl.valueChanges.pipe(
+    startWith(this.searchControl.value),
+    filter((a) => !!a && a.length >= 3),
+    debounceTime(300),
+    distinctUntilChanged()
   );
 
-  ngrxOnStoreInit() {
-    const savedJSONState = localStorage.getItem(PHOTO_STATE_KEY);
-    if (savedJSONState === null) {
-      this.setState(initialState);
-    } else {
-      const savedState = JSON.parse(savedJSONState);
-      this.setState({
-        ...initialState,
-        search: savedState.search,
-        page: savedState.page,
-      });
-    }
-  }
+  readonly searchResults$ = this.search$.pipe(
+    map((search) => ({ search: search || '', page: 1 })), // TypeScript thinks it's string | null
+    switchMap(({ search, page }) =>
+      this.photoService.searchPublicPhotos(search, page).pipe(
+        tap(() => {
+          localStorage.setItem(
+            PHOTO_STATE_KEY,
+            JSON.stringify({ search, page })
+          );
```

Comment:
I know you don't like effect, put localstorage is a good candidate for effect, no ? I would love your opinion on that ?

Reply:
Doesn't effect just subscribe to the observable? This already was doing this with a tapResponse. Side-effects can happen inside taps. I just modified tapResponse, so I didn't really change this very much. I just moved it outside of effect and got something else to subscribe to it instead.

Reply:
I meant signal effect.

Reply:
I didn't think about that. I just modified the code from before. But maybe it wouldn't work there because at the end `injectAutoSignal` resets to initial state, and I wouldn't want an effect to override localStorage back to that.